cooper/constrained_optimizer.py

```diff
@@ -198,6 +232,12 @@ def step(
             if self.cmp.is_constrained:
                 self.dual_step()
 
+                if self.dual_scheduler is not None:
+                    # Do a step on the dual scheduler after the actual step on
+                    # the dual parameters. Extrapolation steps are not followed
```

Comment:
This means that the "intermediate" updates that take place *inside* the extrapolation process do _not_ perform a call to the scheduler's step method. Correct?